gopass_autotype.py

```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/env python3
+
+import json
+import os
+import subprocess
+import sys
+import time
+from os.path import expanduser
+
+home = expanduser("~")
+my_env = os.environ.copy()
+my_env['PATH'] = '/usr/local/bin:{}'.format(my_env['PATH'])
+
+special_autotype_handlers = {
+    ":tab": lambda: do_stroke(48),
+    ":space": lambda: do_stroke(49),
+    ":enter": lambda: do_stroke(36),
+    ":delete": lambda: do_stroke(51),
+    ":delay": lambda: time.sleep(1),
+    ":clearField": lambda: clear_field_content()
+}
+
+
+def get_additional_autotype_handlers_filename():
+    base_path = os.getenv("XDG_CONFIG", home)
+    return '{base_path}/{filename}'.format(base_path=base_path, filename=".gopass_autotype_handlers.json")
+
+
+def load_additional_autotype_handlers_config():
+    handlers_path = get_additional_autotype_handlers_filename()
+    if os.path.exists(handlers_path):
+        with open(handlers_path) as f:
+            return json.load(f)
+    else:
+        return {}
+
+
+def execute_additional_autotype_handler(command):
+    command_result = os.popen(command).read()
+    do_type(command_result)
+
+
+def load_additional_autotype_handlers():
+    config = load_additional_autotype_handlers_config()
+    return dict(map(lambda item: (item[0], lambda: execute_additional_autotype_handler(item[1])), config.items()))
+
+
+def all_autotype_handlers():
+    additional = load_additional_autotype_handlers()
+    return {**additional, **special_autotype_handlers}
+
+
+def do_type(to_type):
+    os.system(f"echo 'tell application \"System Events\" to keystroke \"{to_type}\"' | osascript")
+
+
+def do_stroke(stroke):
+    os.system(f"echo 'tell application \"System Events\" to key code \"{stroke}\"' | osascript")
+
+
+def clear_field_content():
+    os.system("""echo 'tell application "System Events" to keystroke "a" using command down' | osascript""")
+    do_stroke(51)
```

Comment:
I'm not convinced how this is actually clearing a field? Isn't it just adding 51 times a char `a` to the field?

Reply:
Well this is a bit special - how do you clear a field only from the command line? What I came up with is just issue a Ctrl+a and issue a backspace. 
You can find the full reference of available keystrokes [here](http://macbiblioblog.blogspot.com/2014/12/key-codes-for-function-and-special-keys.html).